cognitedata/reveal
# feat: add average framerate user metrics

#### Type of change

![Feat](https://img.shields.io/badge/Type-Feat-green)  

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/REV-742

## Description :pencil:

Adds new Mixpanel event called `sessionEnded` that is sent together with fps value and session length in milliseconds when user ends reveal "session". By "session", I mean time from opening/"resuming" a page with reveal viewer and closing the tab/switching to other tab/navigating to a different part of the same page where reveal window is not displayed anymore.


## How has this been tested? :mag:

Tested with local documentation on desktop (Chrome, Safari, Firefox) and mobile (Safari, iOS).



## Test instructions :information_source:
1. Open documentation preview link and load any example with live viewer.
2. Navigate around the viewer a little bit to "gather" fps data.
3. Open Chrome DevTools -> Network tab.
4. Switch to another tab and back or go to different documentation example preserving network log.
5. Notice that there is new request sent to mixpanel, you can also check fps value inspecting request payload.




## Checklist :ballot_box_with_check:
